src/Integration.Vsix/Analysis/AnalysisStatusNotifier.cs

```diff
@@ -0,0 +1,80 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using SonarLint.VisualStudio.Core.Analysis;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.Analysis
+{
+    [Export(typeof(IAnalysisStatusNotifier))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class AnalysisStatusNotifier : IAnalysisStatusNotifier
+    {
+        private readonly IVsStatusbar vsStatusBar;
+
+        [ImportingConstructor]
+        public AnalysisStatusNotifier([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+        {
+            vsStatusBar = serviceProvider.GetService(typeof(IVsStatusbar)) as IVsStatusbar;
+        }
+
+        public void AnalysisStarted(string filePath)
+        {
+            ShowSpinner();
+            ShowStatus(AnalysisStrings.Notifier_AnalysisStarted, filePath);
+        }
+
+        public void AnalysisFinished(string filePath)
+        {
+            HideSpinner();
+            ShowStatus(AnalysisStrings.Notifier_AnalysisEnded, filePath);
+        }
+
+        public void AnalysisCancelled(string filePath)
+        {
+            HideSpinner();
+            ShowStatus(AnalysisStrings.Notifier_AnalysisCancelled, filePath);
+        }
+
+        public void AnalysisFailed(string filePath)
+        {
+            HideSpinner();
+            ShowStatus(AnalysisStrings.Notifier_AnalysisFailed, filePath);
+        }
+
+        private void ShowStatus(string text, params object[] args)
+        {
+            vsStatusBar.SetText(string.Format(text, args));
+        }
+
+        private void ShowSpinner()
+        {
+            vsStatusBar.Animation(1, Microsoft.VisualStudio.Shell.Interop.Constants.SBAI_General);
```

Comment:
I can't get the damn spinner to show!
It returns an error code `-2147024809` which I can't find anywhere.
I've been following the [documentation](https://docs.microsoft.com/en-us/visualstudio/extensibility/extending-the-status-bar?view=vs-2015&redirectedfrom=MSDN), I don't know what I'm missing. I turned off "Automatically adjust visual experience based on client performance" like they suggested, and I still get an error code :( 
Do you have an idea why?

Reply:
Try using the sample code code from the VS docs: https://docs.microsoft.com/en-us/visualstudio/extensibility/extending-the-status-bar?view=vs-2019#display-an-animation

The fact that the sample code is creating an object first does make a difference in this case.

Then, you'll find it throws an exception because you aren't on the UI thread. ;-)

Reply:
Hooray, it works! Thanks!